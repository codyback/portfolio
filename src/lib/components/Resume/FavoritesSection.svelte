<script lang="ts">
	import type { Homepage } from '@prisma/client';

	export let homepage: Homepage;
</script>

<section
	id="cvFavorites"
	class="mt-10 sm:mt-14 text-sm font-bold uppercase dark:text-neutral-300"
>
	<div
		class="border-t-2 border-black dark:border-neutral-500 grid grid-cols-2 py-3"
	>
		<div class="col-span-1">Favorite Tech</div>
		<div class="col-span-1 text-right">
			<a
				href={homepage.favoriteTechURL}
				rel="noreferrer"
				target="_blank"
				class="hover:underline dark:hover:text-white hover:text-neutral-500"
			>
				{homepage?.favoriteTech}
			</a>
		</div>
	</div>
	<div
		class="border-t-2 border-black dark:border-neutral-500 grid grid-cols-2 py-3"
	>
		<div class="col-span-1">Learning</div>
		<div class="col-span-1 text-right">
			<a
				href={homepage?.currentlyLearningURL}
				rel="noreferrer"
				target="_blank"
				class="hover:underline dark:hover:text-white hover:text-neutral-500"
			>
				{homepage?.currentlyLearning}
			</a>
		</div>
	</div>
	<div
		class="border-t-2 border-b-2 border-black dark:border-neutral-500 grid grid-cols-2 py-3"
	>
		<div class="col-span-1">Reading</div>
		<div class="col-span-1 text-right">
			<a
				href={homepage?.currentlyReadingURL}
				rel="noreferrer"
				target="_blank"
				class="hover:underline dark:hover:text-white hover:text-neutral-500"
			>
				{homepage?.currentlyReading}
			</a>
		</div>
	</div>
	<div class="mt-10">
		<button
			class="py-3 w-full border-2 border-black dark:border-gray-500 dark:hover:border-gray-200 dark:text-white transition-all duration-300 dark:hover:text-black uppercase text-sm font-bold hover:bg-neutral-200"
			>Send Message</button
		>
	</div>
</section>
