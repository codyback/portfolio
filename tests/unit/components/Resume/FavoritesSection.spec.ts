import FavoritesSection from '@/lib/components/Resume/FavoritesSection.svelte';
import type { Homepage } from '@prisma/client';
import { render } from '@testing-library/svelte';
import { describe, it, expect } from 'vitest';

const homepage: Homepage = {
	id: 'home-id',
	profession: 'profession',
	tagline: 'tagline',
	description:
		'Cupidatat et minim sit culpa irure exercitation do laboris veniam id duis eu aute enim.',
	favoriteTech: 'favorite-tech',
	favoriteTechURL: 'favorite-tech-url',
	currentlyLearning: 'currently-learning',
	currentlyLearningURL: 'currently-learning-url',
	currentlyReading: 'currently-reading',
	currentlyReadingURL: 'currently-reading-url',
};

describe('CertificatesSection', () => {
	it('renders correctly with open false', () => {
		const { container } = render(FavoritesSection, { homepage });

		expect(container).toMatchSnapshot();
	});
});
