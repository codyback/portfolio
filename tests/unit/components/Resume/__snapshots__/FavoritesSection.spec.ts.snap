// Vitest Snapshot v1

exports[`CertificatesSection > renders correctly with open false 1`] = `
<body>
  <div>
    <section
      class="mt-10 sm:mt-14 text-sm font-bold uppercase dark:text-neutral-300"
      id="cvFavorites"
    >
      <div
        class="border-t-2 border-black dark:border-neutral-500 grid grid-cols-2 py-3"
      >
        <div
          class="col-span-1"
        >
          Favorite Tech
        </div>
         
        <div
          class="col-span-1 text-right"
        >
          <a
            class="hover:underline dark:hover:text-white hover:text-neutral-500"
            href="favorite-tech-url"
            rel="noreferrer"
            target="_blank"
          >
            favorite-tech
          </a>
        </div>
      </div>
       
      <div
        class="border-t-2 border-black dark:border-neutral-500 grid grid-cols-2 py-3"
      >
        <div
          class="col-span-1"
        >
          Learning
        </div>
         
        <div
          class="col-span-1 text-right"
        >
          <a
            class="hover:underline dark:hover:text-white hover:text-neutral-500"
            href="currently-learning-url"
            rel="noreferrer"
            target="_blank"
          >
            currently-learning
          </a>
        </div>
      </div>
       
      <div
        class="border-t-2 border-b-2 border-black dark:border-neutral-500 grid grid-cols-2 py-3"
      >
        <div
          class="col-span-1"
        >
          Reading
        </div>
         
        <div
          class="col-span-1 text-right"
        >
          <a
            class="hover:underline dark:hover:text-white hover:text-neutral-500"
            href="currently-reading-url"
            rel="noreferrer"
            target="_blank"
          >
            currently-reading
          </a>
        </div>
      </div>
       
      <div
        class="mt-10"
      >
        <button
          class="py-3 w-full border-2 border-black dark:border-gray-500 dark:hover:border-gray-200 dark:text-white transition-all duration-300 dark:hover:text-black uppercase text-sm font-bold hover:bg-neutral-200"
        >
          Send Message
        </button>
      </div>
    </section>
    <!--&lt;FavoritesSection&gt;-->
  </div>
</body>
`;
